addon/modifiers/track-on-deletion.ts

```diff
@@ -0,0 +1,37 @@
+import Modifier from 'ember-modifier';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+import ActivityTracking, { Activity } from '../services/activity-tracking';
+
+interface TrackOnDeletionModifierArgs {
+  positional: [string];
+  named: {
+    relative: boolean;
+  };
+}
+
+export default class TrackOnDeletionModifier extends Modifier<TrackOnDeletionModifierArgs> {
+  @service declare activityTracking: ActivityTracking;
+
+  get description(): string {
+    return this.args.positional[0];
+  }
+
+  willDestroy() {
+    assert(
+      '[modifier][track-on-deletion] An actionDescription needs to be passed for the activity-log to make sense.',
+      typeof this.description === 'string'
+    );
+    const activity: Activity = this.buildActivityObject();
+    this.activityTracking.log(activity);
+  }
+
+  private buildActivityObject(): Activity {
+    return {
+      type: 'component_view',
```

Comment:
can we make sure the event type are well defined

Reply:
This one is meant for modals for instance, they are not pages and they are not button clicks so I took the liberty of defining a new type. As I said above, I wasn't sure the BE had started on this, I'll sync with Karim after I sync with you :)